### PR TITLE
Add text generation pipeline

### PR DESCRIPTION
**Pull request by dacorvo**
This simply adds the text-generation task to the list of tasks supported for `optimum-neuron` pipelines.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import pytest
```
Comment (JingyaHuang) [inline tests/conftest.py:1]: Add copyright text?

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    "flaubert",
+    "mobilebert",
+    "mpnet",
+    "roberta",
+    "roformer",
+    "xlm",
+    "roberta",
+]
+DECODER_ARCHITECTURES = ["gpt2", "llama"]
+DIFFUSER_ARCHITECTURES = ["stable-diffusion", "stable-diffusion-xl"]
```
Comment (JingyaHuang) [inline tests/conftest.py:21]:
```suggestion
DIFFUSION_ARCHITECTURES = ["stable-diffusion", "stable-diffusion-xl"]
```
diffusers is the library name, I think diffusion could be more appropriate.

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+def inf_encoder_model(request):
+    return request.param
+
+
+@pytest.fixture(scope="module", params=[INFERENTIA_MODEL_NAMES[model_arch] for model_arch in DECODER_ARCHITECTURES])
+def inf_decoder_model(request):
+    return request.param
+
+
+@pytest.fixture(scope="module", params=[INFERENTIA_MODEL_NAMES[model_arch] for model_arch in DIFFUSER_ARCHITECTURES])
```
Comment (JingyaHuang) [inline tests/conftest.py:56]:
```suggestion
@pytest.fixture(scope="module", params=[INFERENTIA_MODEL_NAMES[model_arch] for model_arch in DIFFUSION_ARCHITECTURES])
```

File: tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+def inf_diffuser_model(request):
```
Comment (JingyaHuang) [inline tests/conftest.py:57]:
```suggestion
def inf_diffusion_model(request):
```

File: tests/pipelines/conftest.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+@pytest.fixture(scope="module", params=STD_TEXT_TASKS)
+def std_text_task(request):
+    return request.param
+
+
+@pytest.fixture(scope="module")
+@requires_neuronx
+def inf_decoder_path(inf_decoder_model):
```
Comment (JingyaHuang) [inline tests/pipelines/conftest.py:40]: When is the fixture used?

**Review by JingyaHuang — APPROVED** [review]
Looks good to me, just left some small nits.

P.S. Thanks for improving the tests ❤️

File: tests/pipelines/conftest.py (hunk 5 above)
Comment (dacorvo) [reply tests/pipelines/conftest.py:40]: It is not ... yet. Basically it will be used when I add the tests to instantiate the pipelines with local models.

**Pull request merged**